feat(live 23727): spendable rewards

alevito opened the pull request:
Copied from @pavanvora's https://github.com/LedgerHQ/ledger-live/pull/12362
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already. Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### ✅ Checklist

<!-- Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready. -->

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** <!-- if not, please explain. (Feature must be tested / Bug fix must bring non-regression) -->
- [x] **Impact of the changes:** <!-- Please take some time to list the impact & what specific areas Quality Assurance (QA) should focus on -->
  - `spendableBalance` field is only used for UI purpose so it shouldn't affect anything imp

### 📝 Description

In Cardano, staking rewards can be withdrawn and used within a single transaction if the account is delegated to a dRep. However, the `spendableBalance` field was previously ignoring rewards in its calculation.
This PR updates the logic to include staking rewards in `spendableBalance` when the account is already delegated to a dRep.

<!--
| Before        | After         |
| ------------- | ------------- |
|               |               |
-->

### ❓ Context

- **JIRA or GitHub link**: <!-- Attach the relevant ticket number if applicable. (e.g., [JIRA-123] for Jira or #123 for a Github issue) -->


---

### 🧐 Checklist for the PR Reviewers

<!-- Please do not edit this if you are the PR author -->

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

francois-guerin-ledger commented on libs/coin-modules/coin-cardano/src/synchronisation.unit.test.ts:
```diff
@@ -126,31 +109,102 @@ describe("makeGetAccountShape", () => {
         } as any),
       );
+      getDelegationInfoMock.mockReturnValue(
+        Promise.resolve({
```
Same, need to change it everywhere
```suggestion
      getDelegationInfoMock.mockResolvedValue({
```

alevito replied:
changed in latest commit

francois-guerin-ledger commented on libs/coin-modules/coin-cardano/src/synchronisation.unit.test.ts:
```diff
@@ -126,31 +109,102 @@ describe("makeGetAccountShape", () => {
+
+    it("should return spendable balance with rewards when delegated to dRep", async () => {
+      getTransactionsMock.mockReturnValue(
+        Promise.resolve({
```
More direct
```suggestion
      getTransactionsMock.mockResolvedValue({
```

francois-guerin-ledger commented:
Need to rebase on `origin/develop` for Detox fix

alevito commented:
@francois-guerin-ledger , I've rebased this branch on `origin/develop`, and updated all the `mockReturnValue(Promise.resolve(...))` to `mockResolvedValue(...)`.
Would you mind reviewing again please?
